python/ray/tests/test_scheduling_2.py

```diff
@@ -690,7 +694,14 @@ def check_backlog_info():
 
     # In ASAN test it's slow.
     # Wait for 20s for the cluster to be up
-    wait_for_condition(check_backlog_info, 20)
+    try:
+        wait_for_condition(check_backlog_info, 20)
+    except RuntimeError:
+        tasks = list_tasks(limit=10000)
+        print(f"Total tasks: {len(tasks)}")
+        for task in tasks:
+            print(task)
```

Comment:
can we print fewer than 10000? this is going to be a lot of output spam

Reply:
Actually, it should print only 1000 tasks due to my change 
<img width="608" alt="image" src="https://github.com/user-attachments/assets/75dce925-db42-4e8f-8e11-97f18953fdd0" />
I think we are safe to lower the number to something like 1010. We still need to query more than 1000 tasks to see if there are unexpected tasks in the cluster causing the flakiness.